src/mon/OSDMonitor.cc

```diff
@@ -7902,31 +7902,75 @@ int OSDMonitor::get_crush_rule(const string &rule_name,
   return 0;
 }
 
-int OSDMonitor::check_pg_num(int64_t pool, int pg_num, int size, ostream *ss)
```

Comment:
Well, the commit description is misleading – it's not a squash of *all* the commits in the src PR; for instance the `get_num_acting_pgs()` change is missed.

Instead of a squashing & rewriting, it might be desirable to judge the backportability of the dependencies restricting the differences just to C++20.